packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -267,7 +267,10 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     final double leadingScrollOffset = indexToLayoutOffset(itemExtent, firstIndex);
     final double trailingScrollOffset = indexToLayoutOffset(itemExtent, lastIndex + 1);
 
-    assert(firstIndex == 0 || childScrollOffset(firstChild!)! - scrollOffset <= precisionErrorTolerance);
+    // Allow a slightly bigger precision error, because firstIndex already might have
+    // additional precision error, which comes from `getMinChildIndexForScrollOffset`,
+    // which also uses `precisionErrorTolerance`.
+    assert(firstIndex == 0 || childScrollOffset(firstChild!)! - scrollOffset <= precisionErrorTolerance * 10);
```

Comment:
It is weird we multiply the precision error by a constant scaling.

I would do it in the getMinChildIndexForScrollOffset

specifically, by changine the if case to the following
```
      if ((actual * itemExtent - round * itemExtent).abs() < precisionErrorTolerance) {
        return round;
      }
```

Reply:
Ah, ok. I changed `getMaxChildIndexForScrollOffset` aswell for consistency.